actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -1604,6 +1604,9 @@ def fields_for(record_name, record_object = nil, fields_options = {}, &block)
         when String, Symbol
           if nested_attributes_association?(record_name)
             return fields_for_with_nested_attributes(record_name, record_object, fields_options, block)
+          elsif !defined?(@auto_index) && !options[:index] && record_name.to_s.end_with?('[]')
+            record_name = record_name.to_s.sub(/(.*)\[\]$/, "[\\1][#{record_object.id}]")
```

Comment:
perhaps do this line in the blow `if nested` block to keep all string building in one place